### PR TITLE
Add Connect With Project component

### LINKED ISSUE #41: "Connect With Project" widget
screenshot 2018-06-07 16 50 43

### PR DESCRIPTION
**Pull request by rogerhutchings**
Package: app-project

screenshot 2019-01-08 14 58 01

- Adds title and link props to `ClassifyBox`
- Adds the Connect With Project component

Was working on this over Christmas, so am PRing it now.

Closes #41.

## To do:

- Do we have a canonical way of sanitizing URLs? `dompurify` requires `jsdom` to work on the server, and that currently doesn't bundle with webpack, so that's out.
- @beckyrother would you mind reviewing these icons? The icon set used in the designs had some missing, so if you'd like a more consistent look we'll need a different set, or for you to design the extras.

### CODE REVIEW

**Comment by srallen** [comment]
> Do we have a canonical way of sanitizing URLs? dompurify requires jsdom to work on the server, and that currently doesn't bundle with webpack, so that's out.

I don't think we do. `dompurify` worked well to sanitize inputs in the lab when I picked it to resolve the security issue from a few months ago, so I think it can still keep doing that, but maybe there's a better solution to work with strings in react that aren't markdown. 

I'm using [`remark-react`](https://github.com/remarkjs/remark-react) with the rewritten Markdownz and it uses [`hast-util-sanitize`](https://github.com/syntax-tree/hast-util-sanitize) as its default sanitizer. I don't think we want hast nodes here, but remark is built upon the unified.js collective (backed by zeit!) and they have a set of similar utilities for html. This seems to be the parallel utility for sanitization: https://github.com/rehypejs/rehype-sanitize

Since it's a zeit library, we might be able to configure it to work with next.js to just sanitize the current html document that is being rendered? Just hypothesizing here...

As far as icons go, would you consider using grommet icons? We're already using it anyway since it's a peer dep for grommet itself. Even if the grommet icon lib itself doesn't have what we need, we can use the base `Icon` component to wrap the svg to make sure it can be styled like the rest of the theme. Check out the `InfoIcon` I added for the subject metadata modal.

**Comment by rogerhutchings** [comment]
Now using grommet-icons, with a couple of custom ones:

screenshot 2019-01-08 17 41 52

@beckyrother active links are just underlined:

screenshot 2019-01-08 17 45 03

@srallen I'll try out rehype, but just to sanitize this string for now.

**Comment by beckyrother** [comment]
Oh holy cow those are light, what was I thinking! If it's active, please use the darker styles: 
classifier 
and underline on hover. That's teal #005D69 for the tag and #5C5C5C for the social. Sorry!

**Comment by rogerhutchings** [comment]
@srallen I looked into rehype, but it seemed slightly overkill for checking a single URL string - I went with the `validator` lib instead.

File: packages/app-project/package.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
     "react-pose": "~4.0.2",
     "react-progressive-image": "~0.6.0",
     "styled-components": "~4.1.1",
-    "url-parse": "~1.4.4"
+    "url-parse": "~1.4.4",
+    "validator": "^10.10.0"
```
Comment (srallen) [inline packages/app-project/package.json]: Can we add the package using tilde? I couldn't figure out how to default it to that using lerna and npm 😬

**Pull request merged**